client/components/redux/store.js

```diff
@@ -0,0 +1,113 @@
+import { combineReducers, createStore, applyMiddleware } from 'redux'
+import { logger } from 'redux-logger'
+import thunkMiddleware from 'redux-thunk';
+import axios from 'axios'
+
+// ACTION CONSTANTS
+const GET_PRODUCTS = 'GET_PRODUCTS'
+const GET_SINGLE_PRODUCT = 'GET_SINGLE_PRODUCT'
+const GET_CATEGORIES = 'GET_CATEGORIES'
+const GET_SINGLE_CATEGORY = 'GET_SINGLE_CATEGORY'
+
+// ACTION CREATORS
+const getProducts = (products) => ({type: GET_PRODUCTS, products})
+const getSingleProduct = (product) => ({type: GET_SINGLE_PRODUCT, product})
+
+const getCategories = (categories) => ({type: GET_CATEGORIES, categories})
+const getSingleCategory = (category) => ({type: GET_SINGLE_CATEGORY, category})
+
+// STATES
+const productState = {
+  products: [],
+  singleProduct: {}
+}
+
+const categoryState = {
+  categories: [],
+  singleCategory: {}
+}
+
+// REDUCERS
+const productReducer = (state = productState, action) => {
+  switch (action.type) {
+    case GET_PRODUCTS:
+      return {...state, products: [...state.products, ...action.products]}
+    case GET_SINGLE_PRODUCT:
+      // const singleProd = state.products.filter(elem => action.id === elem.id)
+      return {...state, singleProduct: action.product}
+    default:
+      return state;
+  }
+}
+
+const categoryReducer = (state = categoryState, action) => {
+  switch (action.type) {
+    case GET_CATEGORIES:
+      return {...state, categories: [...state.categories, ...action.categories]}
+    case GET_SINGLE_CATEGORY:
+      // const singleCat = state.categories.filter(elem => action.id === elem.id)
+      return {...state, singleCategory: action.category}
+    default:
+      return state;
+  }
+}
+
+// THUNKS
+
+const axios_instance = axios.create({
+  proxy: {
+    host: '127.0.0.1',
+    port: 3000,
```

Comment:
Curious to see how `PORT` will work on heroku. Maybe this can be a .env variable? process.env.port